enterprise/coderd/licenses.go

```diff
@@ -0,0 +1,181 @@
+package coderd
+
+import (
+	"context"
+	"crypto/ed25519"
+	_ "embed"
+	"net/http"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/golang-jwt/jwt/v4"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/coderd"
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/httpapi"
+	"github.com/coder/coder/coderd/rbac"
+	"github.com/coder/coder/codersdk"
+)
+
+const (
+	CurrentVersion        = 3
+	HeaderKeyID           = "kid"
+	AccountTypeSalesforce = "salesforce"
+	VersionClaim          = "version"
+	PubSubEventLicenses   = "licenses"
+)
+
+var ValidMethods = []string{"EdDSA"}
+
+//go:embed keys/2022-08-12
+var key20220812 []byte
+
+var keys = map[string]ed25519.PublicKey{"2022-08-12": ed25519.PublicKey(key20220812)}
```

Comment:
Could we make a `coderd.API` struct that contains this? Similar to how the AGPL `coderd` package is structured.

Reply:
the public keys are baked in at the package level this way.  Isn't this what we want?

It might make it easier to do integration testing if we could inject the keys in as `coderd.Options` on creation of the API, so that tests could generate their own keys and then use them, but I worry that sort of machinery would be tempting for end users looking to bypass our licensing.  Granted, right now they can just modify the `keys/2022-08-12` file, but each layer we bubble this up through makes it that much more obvious how to get away with it.